diff --git a/conf/battle/monster.conf b/conf/battle/monster.conf
index 7adb358..63c5170 100644
--- a/conf/battle/monster.conf
+++ b/conf/battle/monster.conf
@@ -318,4 +318,8 @@ mob_unlock_time: 2000
 // Defines the time (in ms) after which a monster unlocks its target when trapped.
 // During this time monsters will still be in idle mode and use idle skills on random
 // targets, but they continue chasing their original target when no longer trapped.
 mob_unlock_time: 2000
+
+// Show Element Info [Hyroshima]
+// 0= Off, 1= On
+mob_ele_view: 1
\ No newline at end of file
diff --git a/src/custom/battle_config_init.inc b/src/custom/battle_config_init.inc
index e47aca9..e935ed3 100644
--- a/src/custom/battle_config_init.inc
+++ b/src/custom/battle_config_init.inc
@@ -10,4 +10,7 @@
  * Format:
  *	 { "name", &battle_config.<variable name>, <default value>, <minimum value>, <maximum value> },
  **/
+ 
+	// Show Element Info [Hyroshima]
+	{ "mob_ele_view",                       &battle_config.mob_ele_view,                    0,      0,      1,              },
 
diff --git a/src/custom/battle_config_struct.inc b/src/custom/battle_config_struct.inc
index 017d99d..409184a 100644
--- a/src/custom/battle_config_struct.inc
+++ b/src/custom/battle_config_struct.inc
@@ -12,3 +12,5 @@
  *	 <datatype> name;
  **/
 
+	// Show Element Info [Hyroshima]
+	int mob_ele_view;
\ No newline at end of file
diff --git a/src/map/clif.cpp b/src/map/clif.cpp
index 103a6fa..6f3f818 100644
--- a/src/map/clif.cpp
+++ b/src/map/clif.cpp
@@ -10176,6 +10176,41 @@ void clif_name( struct block_list* src, struct block_list *bl, send_target targe
 #if PACKETVER_MAIN_NUM >= 20180207 || PACKETVER_RE_NUM >= 20171129 || PACKETVER_ZERO_NUM >= 20171130
 				unit_data *ud = unit_bl2ud(bl);
 
+				int ele_group = -1;
+				char ele_name[8] = {0};
+				char output[16] = {0};
+
+				// Show Element Info [Hyroshima]
+				if(battle_config.mob_ele_view)
+				{
+					switch(md->status.def_ele)
+					{
+						case 0: ele_group = 51; strcpy(ele_name,"Neutral");	break;
+						case 1: ele_group = 52; strcpy(ele_name,"Water"); break;
+						case 2: ele_group = 53; strcpy(ele_name,"Earth"); break;
+						case 3: ele_group = 54; strcpy(ele_name,"Fire"); break;
+						case 4: ele_group = 55; strcpy(ele_name,"Wind"); break;
+						case 5: ele_group = 59; strcpy(ele_name,"Poison"); break;
+						case 6: ele_group = 57; strcpy(ele_name,"Holy"); break;
+						case 7: ele_group = 58; strcpy(ele_name,"Shadow"); break;
+						case 8: ele_group = 59; strcpy(ele_name,"Ghost"); break;
+						case 9: ele_group = 60; strcpy(ele_name,"Undead"); break;
+						default: break;
+					}
+
+					if(ud != nullptr)
+					{
+						if(ele_group > -1)
+							md->ud.group_id = ele_group;
+						
+						if(strlen(ele_name))
+						{
+							safesnprintf(output,sizeof(output),"%s Lv: %d",ele_name,md->status.ele_lv);
+							safestrncpy(ud->title, output, NAME_LENGTH);
+						}
+					}
+				}
+
 				if (ud != nullptr) {
 					memcpy(packet.title, ud->title, NAME_LENGTH);
 					packet.groupId = ud->group_id;
